odilia/src/state.rs

```diff
@@ -36,6 +40,30 @@ pub struct ScreenReaderState {
 	pub cache: Arc<Cache>,
 }
 
+pub trait CacheProvider {
```

Comment:
Our `Handler` trait is generic over any state `S`, where `S: Clone`, just needed an additional bound saying it can provide us with a `Cache` value via a trait.